src/index.js

```diff
@@ -225,6 +226,10 @@ class RouterImpl extends React.PureComponent {
         navigate: (to, options) => navigate(resolve(to, uri), options)
       };
 
+      if (onPrivateRoute && element.props.private) {
+        onPrivateRoute(props.navigate);
+      }
+
```

Comment:
I think you need to treat the function as if it would return a Boolean describing the bailout:

```js
// peasant javascript
if(typeof onBeforeRender === 'function' && !onBeforeRender(props) ){
 return ''
}
```